contracts/bonding/BondingManager.sol

```diff
@@ -527,10 +528,29 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
 
         if (_amount > 0) {
             // Transfer the LPT to the Minter
-            livepeerToken().transferFrom(msg.sender, address(minter()), _amount);
+            livepeerToken().transferFrom(_owner, address(minter()), _amount);
```

Comment:
We should be using `msg.sender` instead of `_owner` here since tokens are pulled from `msg.sender` and staked for `_owner`. Otherwise, the tokens would need to be pulled from `_owner` which doesn't work for the use case where a contract holds LPT and is staking it on behalf of another user (i.e. what the L2Migrator is supposed to do).